keps/sig-auth/4872-harden-kubelet-cert-validation/README.md

```diff
@@ -0,0 +1,389 @@
+    plane? No. But requires restarting the kube-apiserver.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No.
+
+###### Does enabling the feature change any default behavior?
+
+Yes. If a cluster is using custom kubelet serving certificates that don't follow the same convention as kubernetes issued certificates (CN is `system:node:<node-name>`),
+enabling this feature will make any connection initiated by the kube-api server fail (logs, exec and port-forwarding).
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled once enabled by just setting the command-line flag to true.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+You just get back the new behavior with the extra cert validation, no extra considerations needed.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will add integration tests to validate the enablement/disablement flow. Test cases specified in a previous section.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail if the feature flag is not enabled but the command-line flag is set.
+
+Already running workloads won't be impacted but cluster users won't be able to access the control plane if the cluster is single-node.
+
+###### What specific metrics should inform a rollback?
+
+Not applicable.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No. There is no data stored for this feature which persists between upgrade / downgrade, or between enable / disable.
+The feature is purely an API server configuration option.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The cluster administrators can check the flags passed to the kube-apiserver if they have access to the control plane nodes.
+If the `--disable-kubelet-cert-cn-validation` flag is not set or set to false, the feature is being used.
+Alternatively the can check the `kubernetes_feature_enabled` metric.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other
+  - Details: users can create a Node with a kubelet serving certificate that doesn't meet the CN requirements enforced by this validation (something different than `system:node:<node-name>`).Then run `kubectl logs` for any pod running in that node. If it returns an error for an invalid certificate, the feature is working.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The average `apiserver_request_duration_seconds` for logs/exec/port-forward requests is within reasonable limits.
```

Comment:
The number of TLS connections made from API server to nodes should not increase as well.